File: Python DS & Algo - Runestone Academy/6. Sorting and Searching/C++/6.4_binary_search.cpp
#include <iostream>
#include <vector>
using namespace std;

bool binary_search(vector<int> arr, int ele, int start, int end){
    if(start > end)
        return false;
    int mid = (start + end) / 2;
    if(arr[mid] == ele)
        return true;
    else if(arr[mid] > ele)
        return binary_search(arr, ele, start, mid - 1);
    else
        return binary_search(arr, ele, mid + 1, end);
}

int main(void){
    vector <int> arr;
    for(int i = 1; i < 50; i+=2)
        arr.push_back(i);
    cout << binary_search(arr, 32, 0, arr.size());
}

File: Python DS & Algo - Runestone Academy/6. Sorting and Searching/6.4_binary_search.py
def binary_search(lst, ele, start, end):
    if start > end:
        return False
    mid = (start + end) // 2
    if lst[mid] == ele:
        return True
    elif lst[mid] > ele:
        return binary_search(lst, ele, start, mid - 1)
    else:
        return binary_search(lst, ele, mid + 1, end)


lst = [1, 3, 5, 7, 9, 11, 13]
ele = 94
print(binary_search(lst, ele, 0, len(lst)-1))
